src/Microsoft.AspNetCore.JsonPatch/Internal/DynamicObjectAdapter.cs

```diff
@@ -0,0 +1,211 @@
+                return false;
+            }
+
+            errorMessage = null;
+            return true;
+        }
+
+        public bool TryRemove(
+            object target,
+            string segment,
+            IContractResolver contractResolver,
+            out string errorMessage)
+        {
+            if (!TryGetDynamicObjectProperty(target, contractResolver, segment, out var property, out errorMessage))
+            {
+                return false;
+            }
+
+            // Setting the value to "null" will use the default value in case of value types, and
+            // null in case of reference types
+            object value = null;
+            if (property.GetType().GetTypeInfo().IsValueType
+                && Nullable.GetUnderlyingType(property.GetType()) == null)
+            {
+                value = Activator.CreateInstance(property.GetType());
+            }
+
+            if (!TrySetDynamicObjectProperty(target, contractResolver, segment, value, out errorMessage))
+            {
+                return false;
+            }
+
+            errorMessage = null;
+            return true;
+
+        }
+
+        public bool TryReplace(
+            object target,
+            string segment,
+            IContractResolver contractResolver,
+            object value,
+            out string errorMessage)
+        {
+            if (!TryGetDynamicObjectProperty(target, contractResolver, segment, out var property, out errorMessage))
+            {
+                return false;
+            }
+
+            if (!TryConvertValue(value, property.GetType(), out var convertedValue))
+            {
+                errorMessage = Resources.FormatInvalidValueForProperty(value);
+                return false;
+            }
+
+            if (!TrySetDynamicObjectProperty(target, contractResolver, segment, convertedValue, out errorMessage))
+            {
+                return false;
+            }
+
+            errorMessage = null;
+            return true;
+        }
+
+        public bool TryTraverse(
+            object target,
+            string segment,
+            IContractResolver contractResolver,
+            out object nextTarget,
+            out string errorMessage)
+        {
+            if (!TryGetDynamicObjectProperty(target, contractResolver, segment, out var property, out errorMessage))
+            {
+                nextTarget = null;
+                return false;
+            }
+            else
+            {
+                nextTarget = property;
+                errorMessage = null;
+                return true;
+            }
+        }
+
+        private bool TryGetDynamicObjectProperty(
+            object target,
+            IContractResolver contractResolver,
+            string segment,
+            out object value,
+            out string errorMessage)
+        {
+            var jsonDynamicContract = contractResolver.ResolveContract(target.GetType()) as JsonDynamicContract;
```

Comment:
Please stop by so we can discuss this in person. I've commented on this about 4 times and haven't yet heard and answer.